lib/question.js

```diff
@@ -111,6 +111,13 @@ module.exports = {
                     options: data.options || {},
                     broken: hasFatalIssue,
                 };
+                if (question.workspace_image !== null) { // if workspace, add graded files to params
+                    variant.params['_workspace_required_file_names'] = question.workspace_graded_files;
+                    if(!('_required_file_names' in variant.params)) {
+                        variant.params['_required_file_names'] = [];
+                    }
+                    variant.params['_required_file_names'].push(...question.workspace_graded_files);
```

Comment:
It's generally safer to use `concat()` so that we don't run into JS limits on number of function arguments (the `...` syntax expands into an argument list). We won't hit that limit here in practice (it's at least 32k in modern browsers) but I prefer to not get in the habit of assuming that arbitrary arrays can expand to argument lists.

Reply:
Done